cores/epoxy/main.cpp

```diff
@@ -15,7 +15,7 @@
  * inserts it into the Serial buffer.
  */
 
-#ifdef EPOXY_DUINO
+#if defined(EPOXY_DUINO) && __unix__
```

Comment:
I expect that the check for EPOXY_DUINO is indeed unnecessary, but the check for __unix__ is necessary given the fact that pretty much everything in that file makes sense only for POSIX systems; Windows just doesn't have termios.

One could argue that the right solution is not to compile the file at all.  That is a little troublesome in the platformio environment.  Platformio likes to compile every source file that it finds inside a library source tree.  There is a way around that, but it is cumbersome.  Guarding code that is inherently and deeply Unix-specific with __unix__ seems appropriate to my way of thinking.

The thing you lose by guarding the entire file is the weak definition of main().  That is not a problem in any situation that already has a main() elsewhere.

Reply:
Don't you need to define a `main()` function under Windows as well? At a minimum, this part:
```C++
  setup();
  while (true) {
    loop();
    yield();
  }
```

Reply:
Indeed!  It turns out that I had that code inside my app guarded by ifdef EPOXY_DUINO.  It would be better to have it in EpoxyDuino itself.

Reply:
I have a patch for the main() thing.  The kicker is that MINGW32 has weak support for weak symbols, to the extent that making main() weak just does not work.  I have a way around it but it is a little ugly so I want to run it by you to see what you think
```
#ifdef __unix__
// Weak reference so that the calling code can provide its own main().
int main(int argc, char** argv)
 __attribute__((weak));

int main(int argc, char** argv) {
  return hostduino_main(argc, argv);
}
#else

// Define EPOXY_DUINO_NO_MAIN if the calling code provides its own main().
# ifndef EPOXY_DUINO_NO_MAIN
int main(int argc, char** argv) {
  return hostduino_main(argc, argv);
}
# endif

#endif
```

Reply:
Since `main.cpp` is a .cpp file, the `EPOXY_DUINO_NO_MAIN` must be defined in the `Makefile` through something like:
```
CPPFLAGS := -D EPOXY_DUINO_NO_MAIN
```
I guess I'm ok with that if you are, since I'm not the user of the MinGW environment.

Quick question on the `__unix__` symbol: Are you sure that all unix environments relevant to EpoxyDuino (i.e. Linux, MacOS, FreeBSD) define that symbol? I wonder if it would better to use `#if ! defined(_WIN32)` (or `#ifdef ! defined(__MINGW32__)`), because it makes grepping for MinGW code easier.